#**1. Data Preprocessing:**

In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
from sklearn.preprocessing import StandardScaler

**Load The Dataset into a Pandas DataFrame**

In [ ]:
data = pd.read_csv('BankCustomerData.csv')
data.head()

age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome  \
0  unknown    5   may       261         1     -1         0  unknown   
1  unknown    5   may       151         1     -1         0  unknown   
2  unknown    5   may        76         1     -1         0  unknown   
3  unknown    5   may        92         1     -1         0  unknown   
4  unknown    5   may       198         1     -1         0  unknown   

  term_deposit  
0           no  
1           no  
2           no  
3           no  
4           no

**Convert categorical variables into dummy variables.**

In [ ]:
df_dummies = pd.get_dummies(data, drop_first=True)


**Handle missing values if any**

In [ ]:
df_dummies.isna().sum()

age                    0
balance                0
day                    0
duration               0
campaign               0
pdays                  0
previous               0
job_blue-collar        0
job_entrepreneur       0
job_housemaid          0
job_management         0
job_retired            0
job_self-employed      0
job_services           0
job_student            0
job_technician         0
job_unemployed         0
job_unknown            0
marital_married        0
marital_single         0
education_secondary    0
education_tertiary     0
education_unknown      0
default_yes            0
housing_yes            0
loan_yes               0
contact_telephone      0
contact_unknown        0
month_jun              0
month_may              0
term_deposit_yes       0
dtype: int64

**Convert the target variable subscribed into a binary format (1 for yes, 0 for no)**

In [ ]:
df_dummies['subscribed'] = df_dummies['term_deposit_yes']

#**2. Feature Selection**

**Decide which features to include in the model. You might exclude highly correlated features to
avoid multicollinearity**


In [ ]:
X = df_dummies.drop(['subscribed','term_deposit_yes'],axis=1)
Y = df_dummies['subscribed']

#**3. Data Splitting:**

**Split the dataset into training and testing sets (typically a 70-30 or 80-20 split).**

In [ ]:
X_train,X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.2, random_state=42)

NameError: name 'train_test_split' is not defined

#**4. Model Training**

**Train a logistic regression model on the training set.**

In [ ]:
model = LogisticRegression()

model.fit(X_train,Y_train)

/usr/local/lib/python3.10/dist-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

In [ ]:
Y_pred = model.predict(X_test)

#**5.Model Evaluation:**

**Evaluate the model's performance on the testing set using metrics such as accuracy, precision,
recall, F1-score, and the confusion matrix**

In [ ]:
accuracy = accuracy_score(Y_test,Y_pred)
conf_matrix = confusion_matrix(Y_test,Y_pred)
class_report = classification_report(Y_test,Y_pred)

In [ ]:
print(f'Accuracy          :   {accuracy}')
print(f'Confusion Matrix  :   {conf_matrix}')
print(f'Class Report      :\n   {class_report}')

Accuracy          :   0.9650706436420722
Confusion Matrix  :   [[2437   22]
 [  67   22]]
Class Report      :
                 precision    recall  f1-score   support

           0       0.97      0.99      0.98      2459
           1       0.50      0.25      0.33        89

    accuracy                           0.97      2548
   macro avg       0.74      0.62      0.66      2548
weighted avg       0.96      0.97      0.96      2548



#**6. Conclusion:**

The model scores 96.51% which indicates that it correctly predicts the target variable around 96.51% of the time the model also gives us the result of true negatives which is 2437, false positives which is 22, false negative which is 67, true positives which is 22,